pg_shard--1.0.sql

```diff
@@ -77,3 +77,78 @@ CREATE FUNCTION worker_copy_shard_placement(table_name text, source_node_name te
 RETURNS void
 AS 'MODULE_PATHNAME'
 LANGUAGE C STRICT;
+
+CREATE FUNCTION partition_column_to_node_string(table_oid oid)
+RETURNS text
+AS 'MODULE_PATHNAME'
+LANGUAGE C;
+
+COMMENT ON FUNCTION partition_column_to_node_string(oid)
+		IS 'return textual form of distributed table''s partition column';
+
+CREATE FUNCTION sync_table_metadata_to_citus(table_name text) RETURNS VOID
+AS $sync_table_metadata_to_citus$
+	DECLARE
+		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
+		dummy_shard_length CONSTANT bigint := 0;
+	BEGIN
+		-- remove shard placements for table
```

Comment:
I had a few higher-level questions on the upsert functionality:

- We expect this to change if:
1. Someone drops the table, re-creates it, and calls master_create_distributed_table(), and master_create_worker_shards() again. This is what Neustar was doing.

2. Someone only removes the metadata in pgs_distribution_metadata for the table, and re-creates shards using master_create_worker_shards(). 

In case 1, the relationId would have changed, thus the delete should be a no-op. There wouldn't be any existing shards in citusDB's metadata for that new relationId.

In case 2, the partition information might be there, but the shard/shard-placement deletion will be no-ops as the metadata won't exist in pgs_distribution_metadata.

If my understanding is correct, could we remove the delete functionality for now and think about this in a later change? (Maybe as part of the more comprehensive two-way sync item?)

My other question was: If we do a metadata-sync and then call drop-table, do you know if CitusDB complains about adding CASCADE, and if it deletes the entry in pg_dist_partition? (It has this behavior for CitusDB distributed tables, which it should start assuming a pg_shard table is after the metadata sync).